packages/trpc/server/routers/viewer/apps.tsx

```diff
@@ -0,0 +1,237 @@
+            });
+          } else {
+            const appKey = deriveAppDictKeyFromType(app.type, appKeysSchemas);
+            const keysSchema = appKeysSchemas[appKey as keyof typeof appKeysSchemas] || null;
+            filteredApps.push({
+              name: app.name,
+              slug: app.slug,
+              logo: app.logo,
+              type: app.type,
+              title: app.title,
+              description: app.description,
+              enabled: dbData?.enabled || false,
+              keys: keysSchema ? keysSchema.keyof()._def.values : null,
+            });
+          }
+        }
+      }
+      return filteredApps;
+    }),
+  toggle: authedProcedure
+    .input(
+      z.object({
+        slug: z.string(),
+        enabled: z.boolean(),
+      })
+    )
+    .mutation(async ({ ctx, input }) => {
+      if (ctx.session.user?.role !== "ADMIN")
+        throw new TRPCError({
+          code: "UNAUTHORIZED",
+        });
+
+      const { prisma } = ctx;
+
+      const app = await prisma.app.update({
+        where: {
+          slug: input.slug,
+        },
+        data: {
+          enabled: !input.enabled,
+        },
+      });
+
+      // Get app name from metadata
+      const localApps = getLocalAppMetadata();
+      const appMetadata = localApps.find((localApp) => localApp.slug === app.slug);
+
+      // If disabling an app then we need to alert users basesd on the app type
+      if (input.enabled) {
+        if (app.categories.some((category) => category === "calendar" || category === "video")) {
+          const appCredentials = await prisma.credential.findMany({
+            where: {
+              appId: app.slug,
+            },
+            select: {
+              user: {
+                select: {
+                  email: true,
+                  locale: true,
+                },
+              },
+            },
+          });
+
+          Promise.all(
+            appCredentials.map(async (credential) => {
+              const t = await getTranslation(credential.user?.locale || "en", "common");
+
+              if (credential.user?.email) {
+                await sendDisabledAppEmail({
+                  email: credential.user.email,
+                  appName: appMetadata?.name || app.slug,
+                  appType: app.categories,
+                  t,
+                });
+              }
+            })
+          );
+        } else {
+          const eventTypesWithApp = await prisma.eventType.findMany({
+            where: {
+              metadata: {
+                path: ["apps", app.slug as string, "enabled"],
+                equals: true,
+              },
+            },
+            select: {
+              id: true,
+              title: true,
+              users: {
+                select: {
+                  email: true,
+                  locale: true,
+                },
+              },
+              metadata: true,
+            },
+          });
```

Comment:
Other app types will be associated with an event type so we search through the metadata of the event type for the app